src/util/saveArticles.ts

```diff
@@ -291,10 +292,10 @@ async function getModuleDependencies(articleId: string, mw: MediaWiki, downloade
     };
 }
 
-async function processArticleHtml(html: string, downloader: Downloader, mw: MediaWiki, dump: Dump, articleId: string) {
+async function processArticleHtml(html: string, downloader: Downloader, mw: MediaWiki, dump: Dump, articleId: string, zimCreator: ZimCreator) {
     let mediaDependencies: Array<{ url: string, path: string }> = [];
     let doc = domino.createDocument(html);
-    const tmRet = await treatMedias(doc, mw, dump, articleId);
+    const tmRet = await treatMedias(doc, mw, dump, articleId, downloader, zimCreator);
```

Comment:
not sure it's a good idea to carry `downloader` and `zimCreator` 3 levels down into `treatMedias`. Normally, we get a set of links back to upper level then handle them by iterating mediaDependencies [here](https://github.com/openzim/mwoffliner/blob/9ea0be2c9fed9463c9f380627e608af882c6cddf/src/util/saveArticles.ts#L165). This way we don't deal with zim file in `treat*()` functions which follow OOP spirit. I belive the function called `processArticleHtml` should know nothing about the entity like zim, so I feel there's better way to handle this ;)
Could we handle subtitles outside this function?

Reply:
I am not sure I understood this correctly, I agree with point that downloader and zimcreator got dragged 3 level down, but trackEle is tightly coupled with videoEle, so IMO it makes sense to have them in a same place. But le me know if you suggest differently?

Reply:
Probably it does make sense to keep status quo because that would require so much changes to make the order here in `processArticleHtml()`. Let's table this until we get a resources to catch up a technical debt.

Reply:
I reopen this because I'm not happy about that either. We need to architecture this in a smarter way.

Reply:
Reverted the change of dragging `downloader` and `zimcreator` to 3 level down and handled this in `processArticleHtml()` as suggested.